code/src/context/ModalContext.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  ReactElement,
+  ReactNode,
+  createContext,
+  useContext,
+  useRef,
+  useState
+} from "react";
+import Dialog from "@mui/material/Dialog";
+import AppBox from "@/components/app-box/AppBox";
+
+type ModalContextType = {
+  handleOpenModal: (component: ReactElement) => void;
+  handleCloseModal: () => void;
+};
+
+type ModalProviderProps = { children: ReactNode };
+
+const ModalContext = createContext<ModalContextType | null>(null);
+
+const ModalProvider = ({ children }: ModalProviderProps) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const modalRef = useRef<ReactElement | null>(null);
```

Comment:
We use two parallel states. I think we can use only one and derive `isModalOpen`, for example like following:

```jsx
const [modal, setModal] = useState<ReactElement | null>(null)

const isModalOpen = Boolean(modal);
```

Reply:
Yeah, it's a great idea. Thank you!